lib/block_view.go

```diff
@@ -7039,11 +7131,250 @@ func (bav *UtxoView) _connectAcceptNFTBid(
+				BidAmountNanos: txMeta.BidAmountNanos,
+			}
+			bav._setNFTBidEntryMappings(newBidEntry)
+		}
+
+		// Add an operation to the list at the end indicating we've connected an NFT bid.
+		utxoOpsForTxn = append(utxoOpsForTxn, &UtxoOperation{
+			Type:            OperationTypeNFTBid,
+			PrevNFTBidEntry: prevNFTBidEntry,
+		})
+
+		return totalInput, totalOutput, utxoOpsForTxn, nil
+	} else {
+		newBidEntry := &NFTBidEntry{
+			BidderPKID:     bidderPKID.PKID,
+			NFTPostHash:    txMeta.NFTPostHash,
+			SerialNumber:   txMeta.SerialNumber,
+			BidAmountNanos: txMeta.BidAmountNanos,
+		}
+		bav._setNFTBidEntryMappings(newBidEntry)
+		// Okay here's where the fun happens. We are submitting a bid on a Buy Now enabled NFT. We create the bid then we call the
+		// _helpConnectNFTSold to handle the royalty payout logic and such.
+		totalInput, totalOutput, utxoOpsForTxn, err := bav._helpConnectNFTSold(HelpConnectNFTSoldStruct{
+			NFTPostHash: txMeta.NFTPostHash,
+			SerialNumber: txMeta.SerialNumber,
+			BidderPKID: bidderPKID.PKID,
+			BidAmountNanos: txMeta.BidAmountNanos,
+
+			BidderInputs: []*DeSoInput{},//txMeta.BidderInputs,
+
+			BlockHeight: blockHeight,
+			Txn: txn,
+			TxHash: txHash,
+			VerifySignatures: verifySignatures,
+			IsNFTBid: true,
+			IsAcceptNFTBid: false,
+		})
+		if err != nil {
+			return 0, 0, nil, errors.Wrapf(err, "_connectNFTBid: ")
+		}
+		return totalInput, totalOutput, utxoOpsForTxn, nil
+	}
+
+}
+
+type HelpConnectNFTSoldStruct struct {
+	NFTPostHash    *BlockHash
+	SerialNumber   uint64
+	BidderPKID     *PKID
+	BidAmountNanos uint64
+	UnlockableText []byte
+
+	// When an NFT owner accepts a bid, they must specify the bidder's UTXO inputs they will lock up
+	// as payment for the purchase. This prevents the transaction from accidentally using UTXOs
+	// that are used by future transactions.
+	BidderInputs []*DeSoInput
+
+	BlockHeight      uint32
+	Txn              *MsgDeSoTxn
+	TxHash           *BlockHash
+	VerifySignatures bool
+
+	// These both shouldn't be true
+	IsAcceptNFTBid   bool
+	IsNFTBid         bool
+}
+
+func (bav *UtxoView) _helpConnectNFTSold(txMeta HelpConnectNFTSoldStruct) (
+	_totalInput uint64, _totalOutput uint64, _utxoOps []*UtxoOperation, _err error) {
+	if txMeta.IsAcceptNFTBid && txMeta.IsNFTBid || !txMeta.IsAcceptNFTBid && !txMeta.IsNFTBid {
+		return 0, 0, nil, fmt.Errorf("This transaction must be either an AcceptNFTBid txn or a NFTBid txn")
+	}
+	nftKey := MakeNFTKey(txMeta.NFTPostHash, txMeta.SerialNumber)
+	prevNFTEntry := bav.GetNFTEntryForNFTKey(&nftKey)
+	// Get the post entry, verify it exists.
+	nftPostEntry := bav.GetPostEntryForPostHash(txMeta.NFTPostHash)
+
 	existingProfileEntry := bav.GetProfileEntryForPublicKey(nftPostEntry.PosterPublicKey)
 	if existingProfileEntry == nil || existingProfileEntry.isDeleted {
 		return 0, 0, nil, fmt.Errorf(
-			"_connectAcceptNFTBid: Profile missing for NFT pub key: %v %v",
+			"Profile missing for NFT pub key: %v %v",
```

Comment:
Same comment as with _helpDisconnectNFTSold.  Seems no lose to add the function name to the error strings here.